matrix.lua

```diff
@@ -297,6 +296,134 @@ local function irc_formatting_to_weechat_color(s)
     return s
 end
 
+local function get_diff_indices_from_string_x(x, y)
+    -- For html_formatting_and_body_to_weechat_color to extract HTML tags
+    -- Returns a table filled with a start and end index pair for each tag
+    local m = x:len()
+    local n = y:len()
+    local start = 0
+    local memo = {}
+    -- Trim common leading characters, for extra speed
+    while start < m and start < n
+            and x:sub(start+1, start+1) == y:sub(start+1, start+1) do
+        start = start + 1
+    end
+    for i = start, m + 1 do
+        memo[i] = {}
+        for j = start, n + 1 do
+            memo[i][j] = -1
+        end
+    end
+    -- Determine length of the LCS
+    for i = start, m + 1 do
+        for j = start, n + 1 do
+            if i == start or j == start then
+                memo[i][j] = 0
+            elseif x:sub(i,i) == y:sub(j,j) then
+                memo[i][j] = memo[i-1][j-1] + 1
+            else
+                memo[i][j] = math.max(memo[i-1][j], memo[i][j-1])
+            end
+        end
+    end
+    -- Work through the memo table to get the different parts in x
+    local indices = {}
+    while m > start or n > start do
+        if x:sub(m, m) == y:sub(n, n) then
+            m = m - 1
+            n = n - 1
+        elseif n > start and (m == start or memo[m][n-1] > memo[m-1][n]) then
```

Comment:
I'm getting:
```
 Error in http_cb: …/matrix.lua:335: attempt to index field '?' (a nil value)
 stack traceback:
…/matrix.lua:928: in function <…/matrix.lua:924>
```

Reply:
I can't reproduce this. Do you know what input triggers this?